build/build.cake

```diff
@@ -0,0 +1,242 @@
+var buildDir = baseDir + "/build";
+var Solution = baseDir + "/UWP Community Toolkit.sln";
+var toolsDir = buildDir + "/tools";
+
+var binDir = baseDir + "/bin";
+var nupkgDir = binDir + "/nupkg";
+
+var signClientSettings = MakeAbsolute(File("SignClientSettings.json")).ToString();
+var signClientSecret = EnvironmentVariable("SignClientSecret");
+var signClientAppPath = toolsDir + "/SignClient/Tools/netcoreapp2.0/SignClient.dll";
+
+var styler = toolsDir + "/XamlStyler.Console/tools/xstyler.exe";
+var stylerFile = baseDir + "/settings.xamlstyler";
+
+var versionClient = toolsDir + "/nerdbank.gitversioning/tools/Get-Version.ps1";
+string Version = null;
+
+var name = "UWP Community Toolkit";
+var address = "https://developer.microsoft.com/en-us/windows/uwp-community-toolkit";
+
+//////////////////////////////////////////////////////////////////////
+// METHODS
+//////////////////////////////////////////////////////////////////////
+
+void VerifyHeaders(bool Replace)
+{
+    var header = FileReadText("header.txt") + "\r\n";
+    bool hasMissing = false;
+
+    Func<IFileSystemInfo, bool> exclude_objDir =
+        fileSystemInfo => !fileSystemInfo.Path.Segments.Contains("obj");
+
+    var files = GetFiles(baseDir + "/**/*.cs", exclude_objDir).Where(file => 
+    {
+        var path = file.ToString();
+        return !(path.EndsWith(".g.cs") || path.EndsWith(".i.cs") || System.IO.Path.GetFileName(path).Contains("TemporaryGeneratedFile"));
+    });
+
+    Information("\nChecking " + files.Count() + " file header(s)");
+    foreach(var file in files)
+    {
+        var oldContent = FileReadText(file);
+        var rgx = new Regex("^(//.*\r?\n|\r?\n)*");
+        var newContent = header + rgx.Replace(oldContent, "");
+
+        if(!newContent.Equals(oldContent, StringComparison.Ordinal))
+        {
+            if(Replace)
+            {
+                Information("\nUpdating " + file + " header...");
+                FileWriteText(file, newContent);
+            }
+            else
+            {
+                Error("\nWrong/missing header on " + file);
+                hasMissing = true;
+            }
+        }
+    }
+
+    if(!Replace && hasMissing)
+    {
+        throw new Exception("Please run '.\\build.ps1 -target=UpdateHeaders' and commit the changes.");
+    }
+}
+
+//////////////////////////////////////////////////////////////////////
+// DEFAULT TASK
+//////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+    .Description("Clean the output folder")
+    .Does(() =>
+{
+    if(DirectoryExists(binDir))
+    {
+        Information("\nCleaning Working Directory");
+        CleanDirectory(binDir);
+    }
+    else
+    {
+        CreateDirectory(binDir);
+    }
+});
+
+Task("Verify")
+    .Description("Run pre-build verifications")
+    .IsDependentOn("Clean")
+    .Does(() =>
+{
+    VerifyHeaders(false);
+});
+
+Task("Version")
+    .Description("Updates the version information in all Projects")
```

Comment:
It doesn't actually do this, does it? It looks like it only determines what the version will be.
Indeed, NB.GV is designed so that you don't have to update the projects at all just to get a new version.

Reply:
I wasn't sure, I was just adapting the description from the current one in the Build file that uses Nerdbank.GitVersioning.
Since AssemblyInfo.cs doesn't exist in all projects anymore, the current description didn't make sense.